### PR TITLE
[belongs to 0022298] added new interface-method "groupConcat" in ilQueryUtils

### PR DESCRIPTION
**Pull request by lbruening**
I made a change to ilQueryUtils to define a new method named "stringAgg".

Reason:
The mysql-specific command "GROUP_CONCAT" is used at several places in Ilias. To be compatible with other databases, this SQL command should be present in the QueryUtils

### CODE REVIEW

File: Modules/DataCollection/classes/Fields/Reference/class.ilDclReferenceFieldModel.php
Comment (chfsx) [inline Modules/DataCollection/classes/Fields/Reference/class.ilDclReferenceFieldModel.php]: The Manager-Module should not be used in consumer-code outside the setup process or DB-Updates. Consumer code only should use code from ilDBInterface. therefore the ilDBInterface should be extended with your new method. but I'd like to rename this to `groupConcat` instead of `stringAgg`.

I know the `loadModule`mathod is on this interface, too. I just added a `@internal` statement to declare this.

File: Modules/OrgUnit/classes/PathStorage/class.ilOrgUnitPathStorage.php
Comment (chfsx) [inline Modules/OrgUnit/classes/PathStorage/class.ilOrgUnitPathStorage.php]: Same as above

File: Services/Database/classes/QueryUtils/class.ilMySQLQueryUtils.php
Comment (chfsx) [inline Services/Database/classes/QueryUtils/class.ilMySQLQueryUtils.php]: Please use parameter declarations such as `string` or `int`, there is no `type`. And please rename `$a_name` to `$field_name` or something like that to make clear what name exactly is meant. Additionally `$a_order` doesn't seem to be declared in the PHPDoc.

you can write all the PHPDoc on the interface and just put a 
```
/**
 * @inheritdoc
 */
```
above the implementations.

File: Services/Database/classes/QueryUtils/class.ilPostgresQueryUtils.php
Comment (chfsx) [inline Services/Database/classes/QueryUtils/class.ilPostgresQueryUtils.php]: see above

File: Services/Database/classes/QueryUtils/class.ilQueryUtils.php
Comment (chfsx) [inline Services/Database/classes/QueryUtils/class.ilQueryUtils.php]: see above

**Review by chfsx — CHANGES_REQUESTED** [review]
@lbruening thanks for this contribution, this will help to improve the PostgreSQL support. 

please see my comments directly in the code

**Comment by lbruening** [comment]
@chfsx  I have accepted your suggestion and moved the method to ilDB interface and its implementations

**Comment by matthiaskunkel** [comment]
Jour Fixe, 05 FEB 2018: Can be merged after PR804 has been merged.

**Review by chfsx — APPROVED** [review]

**Pull request merged**
